portfolio/src/main/webapp/script.js

```diff
@@ -29,73 +29,138 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
+/**
+ * Composes a higher order function with world
+ * @param {function} f - a function with @param {string} which @returns {function}
+ * @return {function} a composition with 'world' and the result of f
+ */
 function composeWorld(f) {
   return f('world');
 }
 
+/**
+ * Returns a greeting function to a person
+ * @param {string} name - the person to be greeted
+ * @return {function} a function which has @param {string} punctuation
+ *                    which then greets a person with {name} upon being called 
+ */
 function greet(name) {
```

Comment:
is there a reason this returns another function instead of having the punctuation as another parameter, and having the function return the greeter string?